llvm/lib/Transforms/Jeandle/JavaOperationLower.cpp

```diff
@@ -26,6 +26,53 @@ using namespace llvm;
 
 namespace {
 
+static bool removeFunctionFromLLVMUsed(Module &M, Function &F) {
+  GlobalVariable *UsedArray = M.getGlobalVariable("llvm.used");
+  if (!UsedArray)
+    return false;
+
+  ConstantArray *InitArray = cast<ConstantArray>(UsedArray->getInitializer());
+  if (!InitArray) {
+    UsedArray->eraseFromParent();
+    return false;
+  }
+
+  std::vector<Constant *> NewElements;
+  bool found = false;
+
+  // Find all elements to be preserved.
+  for (unsigned i = 0; i < InitArray->getNumOperands(); i++) {
+    Constant *Element = InitArray->getOperand(i);
+    if (Function *Func = dyn_cast<Function>(Element)) {
+      if (Func == &F) {
+        found = true;
+        continue;
+      }
+    }
+    NewElements.push_back(Element);
+  }
+
+  if (!found)
+    return false;
+
+  UsedArray->eraseFromParent();
```

Comment:
It seems we need to invoke the method `eraseFromParent` before the `return` statement?